File: software_design_and_engineering/Original_java/TaskServiceTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;


public class TaskServiceTest {
    String testID = "0123456789";
    String testName = "TaskName<21";
    String testDescription = "Description<51";

    @Test
    public void testTaskService() {
        // This task has the same id ("2223456789") to test the update task service
        // Also I am doing this in one big test rather than separating each test
        Task updateTask = new Task("2223456789", "UpdateName", "UpdateDescription");

        TaskService taskService = new TaskService();
        taskService.addTask("2223456789", testName, testDescription);
        taskService.addTask(testID, testName, testDescription);
        taskService.deleteTask(testID);

        taskService.updateTask(updateTask);
        assertEquals(updateTask.getTaskId(), "2223456789");
        assertEquals(updateTask.getTaskName(), "UpdateName");
        assertEquals(updateTask.getTaskDescription(), "UpdateDescription");
    }


}

File: software_design_and_engineering/Enhanced_python/task_service_test.py
import unittest
from task_service import TaskService
from task import Task


class TestTaskService(unittest.TestCase):
    test_id = "0123456789"
    test_name = "TaskName<21"
    test_description = "Description<51"

    def test_task_service(self):
        update_task = Task("2223456789", "UpdateName", "UpdateDescription")
        task_service = TaskService()
        task_service.add_task("2223456789", self.test_name, self.test_description)
        task_service.add_task(self.test_id, self.test_name, self.test_description)
        task_service.delete_id(self.test_id)

        task_service.update_task(update_task)
        self.assertEquals(update_task.get_task_id(), "2223456789")
        self.assertEquals(update_task.get_task_name(), "UpdateName")
        self.assertEquals(update_task.get_task_description(), "UpdateDescription")
